### DISCUSSION TITLE
Can't get isDirty to work (stays true)

### DISCUSSION THREAD

**Discussion by apedretti** [react-hook-form/react-hook-form#8860] [Q&A] [answered]
Hi beautiful people, 

Case and instructions included in the Codesandbox: https://codesandbox.io/s/condescending-clarke-e918z9

Basically my _isDirty_ is working like _isTouched_. What am I missing?

And in case it helps, my objective is to use the form state to only enable the submitting button if the user has changed anything.

**Comment by leapful** [comment, marked as answer]
@apedretti : You forgot to put default values for "useForm" in order to checking dirty state.

Referrer: https://react-hook-form.com/api/useform/formstate

Working demo: https://codesandbox.io/s/festive-jasper-3hbq08?file=/src/App.js.

**Reply by apedretti** [reply]
Thank you for taking the time to solve and answer this. I didn't want to insist on it as the community is giving so much already as it is, but I still needed to solve it somehow -- so again, thank you.
